hudi-common/src/main/java/org/apache/hudi/metadata/HoodieTableMetadataUtil.java

```diff
@@ -1814,16 +1825,97 @@ private static Comparable<?> coerceToComparable(Schema schema, Object val) {
     }
   }
 
-  private static boolean isColumnTypeSupported(Schema schema, Option<HoodieRecordType> recordType) {
+  private static Integer castToInteger(Object val) {
```

Comment:
What's the reason for returning `Integer` (boxing/unboxing) and not primitve `int`? Is it because you need `null` to represent no value? Same question for other types.

Reply:
yes.